class/ChessGame.php

```diff
@@ -0,0 +1,2339 @@
+            'K' => 'wK',
+            'Q' => 'wQ',
+            'R' => 'wR',
+            'B' => 'wB',
+            'N' => 'wN',
+            'P' => 'wP',
+            'k' => 'bK',
+            'q' => 'bQ',
+            'r' => 'bR',
+            'b' => 'bB',
+            'n' => 'bN',
+            'p' => 'bP',
+        ];
+
+        $tiles = \implode('', \array_reverse(\explode('/', $this->gamestate['fen_piece_placement'])));
+
+        for ($i = 0, $iMax = mb_strlen($tiles); $i < $iMax; ++$i) {
+            $tile = $tiles[$i];
+
+            if (isset($piece_map[$tile])) {
+                $this->board[] = $piece_map[$tile];
+            } elseif (\is_numeric($tile) && $tile >= 1 && $tile <= 8) {
+                for ($j = 0; $j < $tile; ++$j) {
+                    $this->board[] = '00';
+                }
+            } else {
+                #trigger_error("tile='$tile'", E_USER_WARNING); #*#DEBUG#
+                return false; // unexpected character in piece_placement
+            }
+        }
+
+        if (64 != \count($this->board)) {
+            #trigger_error('count(board)=' . count($this->board), E_USER_WARNING); #*#DEBUG#
+            return false; // piece_placement has incorrect number of tiles
+        }
+
+        $this->w_figures = [];
+
+        $this->b_figures = [];
+
+        for ($i = 0; $i < 64; ++$i) {
+            $tile = $this->board[$i];
+
+            $coordinates = $this->boardIndexToCoord($i);
+
+            if ('w' == $tile[0]) {
+                $this->w_figures[] = $tile[1] . $coordinates;
+            } elseif ('b' == $tile[0]) {
+                $this->b_figures[] = $tile[1] . $coordinates;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Convert array to FEN piece placement field.
+     *
+     * Use $this->board to initialize $this->gamestate['fen_piece_placement'].
+     *
+     * @access private
+     */
+    public function board_to_fen_piece_placement()
+    {
+        $rows = [];
+
+        for ($rank = 7; $rank >= 0; --$rank) {
+            $row = '';
+
+            for ($file = 0; $file < 8; ++$file) {
+                $index = 8 * $rank + $file;
+
+                if (!$this->is_empty_tile($index)) {
+                    $tile = $this->board[$index];
+
+                    $piece = 'w' == $tile[0] ? $tile[1] : \mb_strtolower($tile[1]); // 'K','Q','R','B','N' or 'P' (uppercase for white, lowercase for black)
+                } else {
+                    $piece = 'x'; // temporarily mark each empty tile with 'x'
+                }
+
+                $row .= $piece; // append piece symbol to row-string
+            }
+
+            $rows[] = $row;
+        }
+
+        // Concatenate the eight row-strings with the separator '/'.
+
+        // Then replace each string of x's with the string length.
+
+        $this->gamestate['fen_piece_placement'] = \preg_replace_callback('/(x+)/', \create_function('$matches', 'return strlen($matches[1]);'), \implode('/', $rows));
+    }
```

Comment:
Missing change from create_function() to function() from 1.08 Alpha 1
```suggestion
        $this->gamestate['fen_piece_placement'] = preg_replace_callback('/(x+)/', function($matches) {return strlen($matches[1]);}, implode('/', $rows));
    }
```